### PR TITLE
Jetpack Cloud: update copy in the "Fix all threats" dialog

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Update copy that we show inside the "Fix all threats" dialog. The goal is to list the actions we would do to fix the threats.
* Added a [Cancel] button. Before, there was no way to close the dialog.
* Created a utils file to encapsulate functionality related to threat objects.

#### Implementation notes

* Added specific types for threats that can be fixed/ignored. TS is happy about this.

#### Questions

* Should we display the [Fix all] button when there is only one fixable threat? I tend to think we should not, but what if the user has a lot of "ignorable" threats and just one "fixable". That "fixable" threat might get lost in the list and the [Fix all] button could help the user see there is a threat.

#### Testing instructions

* Run this PR
* Select a site with as many threats as possible (hopefully, you will have threats of different types)
* Go to Scan -> Scanner
* Click the [Fix all] button
* Verify the copy makes sense (plural)
* Verify you see a short description of the threat and the action Jetpack is going to do to fix the threat

#### Before

image

#### After

image

### CODE REVIEW

File: client/landing/jetpack-cloud/components/threat-item/utils.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	return 'none';
+}
+
+export const getThreatVulnerability = ( threat: Threat ): string | i18nCalypso.TranslateResult => {
+	switch ( getThreatType( threat ) ) {
+		case 'core':
+			return translate( 'Vulnerability found in WordPress' );
+
+		case 'file':
+			return translate( 'Threat found %(signature)s', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item/utils.ts:43]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Threat found ({{signature/}})' )` *ES Score: 7*

File: client/landing/jetpack-cloud/components/fix-all-threats-dialog/index.tsx [hunk 2]
```diff
@@ -115,6 +116,14 @@ const FixAllThreatsDialog = ( {
...
+						{ firstStep === currentStep && (
+							<Button
+								className="fix-all-threats-dialog__btn fix-all-threats-dialog__btn--cancel"
+								onClick={ onCloseDialog }
+							>
+								{ translate( 'Cancel' ) }
+							</Button>
+						) }
 						<Button primary className="fix-all-threats-dialog__btn" onClick={ onConfirmation }>
 							Fix all threats
```
Comment (enejb) [inline client/landing/jetpack-cloud/components/fix-all-threats-dialog/index.tsx:128]: We should probably also translate this. 
```suggestion
							{ translate( 'Fix all threats' ) }
```

File: client/landing/jetpack-cloud/components/threat-item-header/index.tsx [hunk 3]
```diff
@@ -8,49 +8,79 @@ import { translate } from 'i18n-calypso';
...
 
 	switch ( getThreatType( threat ) ) {
 		case 'core':
-			return translate( 'Infected core file: {{filename/}} ', {
-				components: {
-					filename: <code className="threat-item-header__alert-filename">{ basename }</code>,
-				},
-			} );
+			return isStyled
+				? translate( 'Infected core file: {{filename/}}', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item-header/index.tsx:29]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **8** times:
`translate( 'Infected core file: {{filename/}} ' )` *ES Score: 14*

File: client/landing/jetpack-cloud/components/threat-item-header/index.tsx [hunk 4]
```diff
@@ -8,49 +8,79 @@ import { translate } from 'i18n-calypso';
...
+						components: {
+							filename: <code className="threat-item-header__alert-filename">{ basename }</code>,
+						},
+				  } )
+				: translate( 'Infected core file: %s', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item-header/index.tsx:34]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **8** times:
`translate( 'Infected core file: {{filename/}} ' )` *ES Score: 6*

File: client/landing/jetpack-cloud/components/threat-item-header/index.tsx [hunk 5]
```diff
@@ -8,49 +8,79 @@ import { translate } from 'i18n-calypso';
...
+				  } );
 
 		case 'file':
-			return translate( 'The file {{filename/}} contains a malicious code pattern.', {
-				components: {
-					filename: <code className="threat-item-header__alert-filename">{ basename }</code>,
-				},
-			} );
+			return isStyled
+				? translate( 'The file {{filename/}} contains a malicious code pattern', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item-header/index.tsx:40]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'The file {{filename/}} contains a malicious code pattern.' )` *ES Score: 10*

File: client/landing/jetpack-cloud/components/threat-item-header/index.tsx [hunk 6]
```diff
@@ -8,49 +8,79 @@ import { translate } from 'i18n-calypso';
...
+						components: {
+							pluginSlug: (
+								<span className="threat-item-header__alert-slug">{ extension.slug }</span>
+							),
+							version: (
+								<span className="threat-item-header__alert-version">{ extension.version }</span>
+							),
+						},
+				  } )
+				: translate( 'Vulnerable Plugin: %(pluginSlug)s (version %(version)s)', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item-header/index.tsx:61]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **8** times:
`translate( 'Vulnerable Plugin: {{pluginSlug/}} (version {{version/}})' )` *ES Score: 8*

File: client/landing/jetpack-cloud/components/threat-item-header/index.tsx [hunk 7]
```diff
@@ -8,49 +8,79 @@ import { translate } from 'i18n-calypso';
...
+			return isStyled
+				? translate( 'Vulnerable Theme {{themeSlug/}} (version {{version/}})', {
+						components: {
+							themeSlug: <span className="threat-item-header__alert-slug">{ extension.slug }</span>,
+							version: (
+								<span className="threat-item-header__alert-version">{ extension.version }</span>
+							),
+						},
+				  } )
+				: translate( 'Vulnerable Theme %(themeSlug)s (version %(version)s)', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item-header/index.tsx:78]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **9** times:
`translate( 'Vulnerable Theme {{themeSlug/}} (version {{version/}})' )` *ES Score: 8*

**Comment by rcanepa** [comment]
@enejb @keoshi this PR is ready for review. I changed a bit the copy because what we get from the API doesn't match what we have in our designs. Please let me know if this approach makes sense or we should change it.

**Comment by enejb** [comment]
I think this works well. I also update the (Single) Threat Dialog. To use the latest threat info that we are showing. 

I noticed that the 2 different dialogs look quite a bit different. 

Screen Shot 2020-04-29 at 10 58 38 AM

Vs. 

Screen Shot 2020-04-29 at 10 58 30 AM

I am not sure which one we should go with. cc: @keoshi  (I like the single threat dialog better. :) 
)

Lets update them to be consistent and then lets ship this. It is such a nice improvement.

File: client/landing/jetpack-cloud/components/threat-item/utils.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+export const getThreatFix = ( fixable: ThreatFix ): i18nCalypso.TranslateResult => {
+	switch ( fixable.fixer ) {
+		case 'replace':
+			return translate( 'Jetpack Scan will replace the affected file or directory.' );
+		case 'delete':
+			return translate( 'Jetpack Scan will delete the affected file or directory.' );
+		case 'update':
+			if ( fixable.target ) {
+				return translate( 'Jetpack Scan will update to a newer version (%(version)s).', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/threat-item/utils.ts:74]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( 'Jetpack Scan can update to a newer version (%(version)s).' )` *ES Score: 6*

**Pull request merged**